### ISSUE TITLE
Better feedback for Discrete Allocation

### ISSUE THREAD

**Issue by silvavn** [robertmartin8/PyPortfolioOpt#204]
**Is your feature request related to a problem? Please describe.**
Yes. When we run discrete allocation, if a NaN is found then the allocation fails. We however do not know what security causes that NaN Value

**Describe the solution you'd like**
I would like to know what security causes the NaN value

**Comment by robertmartin8** [comment]
@silvavn Thanks for reaching out!

I'm afraid I don't understand exactly what the issue is. Are there NaNs in the inputs to `DiscreteAllocation` (weights or latest prices?)

**Comment by silvavn** [comment]
I believe that there is some calculation that needs a try catch.
I am inputting valid stocks and values and I get the following error:

```python
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/ubuntu/Faidh/faidh_api.py", line 125, in get_optimized_portfolio
    alloc = popt.get_discrete_allocation(float(content["portfolioSize"]), 0.03)
  File "/home/ubuntu/Faidh/p_optimizer.py", line 32, in get_discrete_allocation
    return da.greedy_portfolio()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/pypfopt/discrete_allocation.py", line 186, in greedy_portfolio
    n_shares = int(weight * self.total_portfolio_value / price)
ValueError: cannot convert float NaN to integer
```

**Comment by robertmartin8** [comment]
This is the segment that seems to be throwing the error:

```python

        for ticker, weight in self.weights:
            price = self.latest_prices[ticker]
            # Attempt to buy the lower integer number of shares
            n_shares = int(weight * self.total_portfolio_value / price)
```

Based on the stack trace, either `weight`, `total_portfolio_value`, or `price` is NaN. Could you please check the `weights` and `latest_prices` you provide to the constructor for `DiscreteAllocation`?

Cheers,
Robert

**Comment by silvavn** [comment]
So the weights are all `nan` for some reason. What I did was add a `try/except`  block around `n_shares =...` returning 0 whenever a `nan` value is found. 

I have also changed the typecast to use an integer division (not that it makes any diff).

**Comment by robertmartin8** [comment]
@silvavn 

Interesting, but the weights come from the output of the portfolio optimisation and shouldn't be NaN, so maybe for now you could drop NaNs from your weights/prices? 

You're probably right that I should add some handling for NaNs; I guess I didn't anticipate that there would be any

Robert

**Comment by robertmartin8** [comment]
Fixed in v1.3.0
